app/src/test/java/uk/ac/ebi/atlas/search/GeneSearchServiceTest.java

```diff
@@ -228,4 +229,26 @@ void exceptionsThrownInParallelTasksAreWrapped() {
                 () -> subject.getCellIdsInExperiments(generateRandomExperimentAccession()))
                 .withCauseInstanceOf(ExecutionException.class);
     }
+
+    @Test
+    void whenGeneIdsExistsThenReturnSetOfCellIds() {
+        var existingGeneId1 = "ExistingGeneId1";
+        var existingGeneId2 = "ExistingGeneId2";
+        var validGeneIds = ImmutableSet.of(existingGeneId1, existingGeneId2);
+        var existingCellId1 = "ExistingCellId1";
+        var existingCellId2 = "ExistingCellId2";
+        var experimentAccession = "ExperimentAccession";
+        var expectedCellIds = ImmutableSet.of(existingCellId1, existingCellId2);
```

Comment:
Have a look at our `RandomDataTestUtils` to generate all these values, so we can keep the invariant as small as possible.